user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.js

```diff
@@ -2022,10 +2030,11 @@ define([
           obj.file.isProviderRootPath = true;
         }
 
+        // TODO BACKLOG-41091: Decoding would ideally be coming from the back-end (see also, BACKLOG-41089, BACKLOG-41229)
         //decode potentially encoded name and title attributes of PVFS paths
-        if( !isRepositoryPath(obj.file.path) && !obj.file.decoded ){
-          obj.file.name = decodePvfsFileAttribute(obj.file.name);
-          obj.file.title = obj.file.title === null ? obj.file.name : decodePvfsFileAttribute(obj.file.title);
+        if (!isRepositoryPath(obj.file.path) && !obj.file.decoded) {
+          obj.file.name = Encoder.urlDecodePvfsFilePath(obj.file.name);
+          obj.file.title = obj.file.title === null ? obj.file.name : Encoder.urlDecodePvfsFilePath(obj.file.title);
```

Comment:
We definitely should have this coming from the server.

Similarly to https://github.com/pentaho/pentaho-kettle/pull/9450:
- the server-side `IGenericFile` interface, would get a new `getNameDecoded()` method whose default implementation would return `getName()`
- the VFS implementation, `VFSFile`, would override to actually decode any `%` characters in `getName()`
- this would get serialized to the client where we'd read the result from the property `nameDecoded`, instead of `name`, where appropriate.
- the GWT `GenericFile` class would get a new method `getNameDecoded()` (and a corresponding `setNameDecoded(.)`)
- certain operations, such as finding a selected path in the folder tree, still need to use `name`.

The title should be ensured to be non-encoded, directly on the server, before calling `setTitle(.)`.

I find that the complexity, difficulty and risk of the current proposed solution is not smaller than the "TODO way" and would prefer we'd follow it instead.